db/db_utils.go

```diff
@@ -243,8 +244,29 @@ func Update(collection string, query interface{}, update interface{}) error {
 	return db.C(collection).Update(query, update)
 }
 
-// Update updates one matching document in the collection.
+// UpdateContext updates one matching document in the collection.
 func UpdateContext(ctx context.Context, collection string, query interface{}, update interface{}) error {
+	// Temporarily, we check if the document has a key beginning with '$', this would
+	// indicate a proper update operation. If not, it's a document intended for replacement.
+	// If the document is unable to be transformed (aka err != nil, e.g. a pipeline), we
+	// also default to an update operation.
+	// This will be removed in DEVPROD-15419.
```

Comment:
This is functionality that comes from the original Update implementation. It's not within scope of this ticket to correct this, but I wanted to add some logging to see where we do it. From my searching, it looks like only in tests (and I corrected those to ReplaceContext).